sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormRecognizerClient.cs

```diff
@@ -318,5 +289,35 @@ public virtual FormTrainingClient GetFormTrainingClient()
         }
 
         #endregion Training client
+
+        /// <summary>
+        /// Used as part of argument validation. Detects the <see cref="ContentType"/> of a stream and
+        /// throws an <see cref="ArgumentException"/> in case of failure.
+        /// </summary>
+        /// <param name="stream">The stream to which the content type detection attempt will be performed.</param>
+        /// <param name="paramName">The original parameter name of the <paramref name="stream"/>. Used to create exceptions in case of failure.</param>
+        /// <returns>The detected <see cref="ContentType"/>.</returns>
+        /// <exception cref="ArgumentException">Happens when detection fails or cannot be performed.</exception>
```

Comment:
I forgot, will .NET provide `contentType` as a parameter on the method for use if auto-detection fails? If so, might be worth mentioning it in the exception.

Reply:
You have a point. For the time being, the `contentType` can be provided in the `RecognizeOptions` instance that's passed to the public methods. I'll get the exception messages updated.